03 mongodb

Veronika-chenko opened the pull request.

Veronika-chenko committed:
feature/mongodb

-support commit

Veronika-chenko committed:
feature/mongodb

help-point

Veronika-chenko committed:
feature/commit

Veronika-chenko committed:
feature/mongodb

-applied mongoose
-added services/contactService.js to interact with Database
-used helpers/apiHelper.js instead of try/catch for controllers
-refactored routes/api/contacts.js with builder notation

KhYulian commented on server.js:
```diff
@@ -1,7 +1,20 @@
+  }
+}
+
+start()
```
Можна скористатись IIFE

Veronika-chenko replied:
застосувала

KhYulian commented on src/controllers/contactsControllers.js:
```diff
@@ -0,0 +1,55 @@
+const getContactByIdController = async (req,res) => {
+    const { contactId } = req.params;
+    const contact = await getContactById(contactId)
+    if (!contact) return res.sendStatus(404) 
```
res.status(404).json({message: "Contact not found"}) - поівдомлення не завадить, аналогічно в інших місцях

Veronika-chenko replied:
повиправляла

KhYulian commented on src/controllers/contactsControllers.js:
```diff
@@ -0,0 +1,55 @@
+const removeContactController = async (req, res) => {
+    const { contactId } = req.params;
+    const contactDelete = await removeContact(contactId)
+    if (!contactDelete) return res.sendStatus(404)
```
використовуйте фігурні дужки після if для кращої читабельності

Veronika-chenko replied:
повиправляла

KhYulian reviewed (changes requested):
Потрібне доопрацювання

KhYulian commented:
Також подавайте в роботу в ЛМС, бо я можу пропустити ваш запит на ревю

Veronika-chenko committed:
feature/mongodb

-applied IIFE for start server fn
-applied curly braces for if statements
-add error messages for res.status(404)

KhYulian reviewed (approved):
Чудова робота ❤️❤️
